controller/lib/core/controller.cpp

```diff
@@ -90,20 +90,19 @@ Controller::Run(Time now, const VentParams &params,
     // them to the desired positions.
     blower_valve_pid_.Reset();
 
-    // Reset the two flow integrators, forcing volume to 0.
-    //
-    // TODO: This isn't ideal; if the blower is off, we should still be able to
-    // report volume.
-    flow_integrator_.emplace();
-    uncorrected_flow_integrator_.emplace();
-
     actuators_state = {
         .fio2_valve = 0,
         .blower_power = 0,
         .blower_valve = 0,
         .exhale_valve = 1,
     };
+    ventilator_was_off_ = true;
```

Comment:
Is it still correct to `NoteExpectedVolume(0)` above when we transition from on -> off or off -> on?  I guess we can hold off on that until we get your flow denoising for off mode.

Reply:
Yeah... I'm not sure either.
I'm currently running some tests on the drift suppression and the two algorithms basically step on each others toes when the vent is off: if there is a flow correction (based on the last breath), my algorithm will make the uncorrected flows go back to zero as soon as everything settles in a constant mode, but the current algorithm will apply its correction on this already zeroed flow, and make the volume diverge again...
So basically we will have to use a lot of caution as to when we should apply a correction and when we shouldn't... For now I think applying the correction makes sense. With the drift suppression, not so sure. I'll run more tests to find the least wrong answer (I don't think there is a perfect solution here...)